### PR TITLE
Moving to Symfony and to DIC

### PR DESCRIPTION
**Pull request by CarolinWickner**

### CODE REVIEW

**Review by mfranczak — CHANGES_REQUESTED** [review]
Please remove cache from git repository.

File: app/config/services.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    # this creates a service per class whose id is the fully-qualified class name
+    AppBundle\:
+        resource: '../../src/AppBundle/*'
+        # you can exclude directories or files
+        # but if a service is unused, it's removed anyway
+        exclude: '../../src/AppBundle/{Entity,Repository,Tests}'
+
+    # controllers are imported separately to make sure they're public
+    # and have a tag that allows actions to type-hint services
+    AppBundle\Controller\:
```
Comment (mfranczak) [inline app/config/services.yml]: In line 10 you removed autowiring and then autoconfigure, therefore this line makes no sense..

File: app/config/services.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        autowire: false
+        # automatically registers your services as commands, event subscribers, etc.
+        autoconfigure: false
+        # this means you cannot fetch services directly from the container via $container->get()
+        # if you need to do this, you can override this setting on individual services
+        public: false
+
+    # makes classes in src/AppBundle available to be used as services
+    # this creates a service per class whose id is the fully-qualified class name
+    AppBundle\:
```
Comment (mfranczak) [inline app/config/services.yml]: In line 10 you removed autowiring and then autoconfigure, therefore this line makes no sense..

File: composer.json [hunk 3]
```diff
@@ -1,22 +1,71 @@
 {
     "name": "caro/aquarium-shop",
-    "description": "Aquarium Shop Coding Exercise",
+    "license": "proprietary",
```
Comment (mfranczak) [inline composer.json]: :)

File: src/AppBundle/Controller/ShopController.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+    * @Route("/index")
+    * @Method("GET")
+    */
+    public function indexAction() {
+        $return_array = [
+            'fishTypes' => $this->repository->getFishTypes(),
+            'aquariums' => $this->repository->getAquariums(),
+        ];
+        header('Content-Type: application/json');
```
Comment (mfranczak) [inline src/AppBundle/Controller/ShopController.php]: there is no need to send this header. JsonResponse object already sets the correct header. Please take I look to the implementation.

File: src/AppBundle/Controller/ShopController.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace AppBundle\Controller;
+
+use AppBundle\Repository\AquariumRepository;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+
+class ShopController extends Controller {
```
Comment (mfranczak) [inline src/AppBundle/Controller/ShopController.php]: There is no need to extend Symfony controller here. You don't use the underlining methods.

File: src/AppBundle/Repository/AquariumRepository.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+use AppBundle\Entity\Aquarium;
+use AppBundle\Entity\FishInAquarium;
+use AppBundle\Entity\FishType;
+use AppBundle\Entity\FloatRange;
+use AppBundle\Entity\Gadget\WaterAbsorbGadget;
+use AppBundle\Entity\Gadget\WaterExchangeGadget;
+use AppBundle\Entity\Gadget\WaterMeasuringGadget;
+
+class AquariumRepository {
```
Comment (mfranczak) [inline src/AppBundle/Repository/AquariumRepository.php:19]: How about breaking down this repository?
Currently you have FishTypes and Aquarium managed in one repository.

File: tests/AppBundle/Controller/DefaultControllerTest.php [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Tests\AppBundle\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class DefaultControllerTest extends WebTestCase
```
Comment (mfranczak) [inline tests/AppBundle/Controller/DefaultControllerTest.php]: ?

**Comment by mfranczak** [comment]
Looks nice, please make sure you don't have any code outside your project and consider breaking the large repository to smaller classes.

File: src/AppBundle/Repository/AquariumRepository.php (hunk 6 above)
Comment (CarolinWickner) [reply src/AppBundle/Repository/AquariumRepository.php:19]: I decided not to break down the repository at this point, I added a comment explaining why.

**Pull request merged**